### ISSUE TITLE
Theres a javascript interpreter for 3ds (CTRJS)

### ISSUE THREAD

**Issue by GamerFox2000** [coderman64/netsurf-3ds#4]
The javascript interpreter can be found on https://github.com/Symbitic/CTRJS

**Comment by Anonymous941** [comment]
This designed homebrew development, not websites, so it would take a while to modify it to work with the HTML.  Also most (as in nearly all) websites with JavaScript are extremely complicated, which would not be feasible for the 3DS to run.  I'm not a maintainer so correct me if I'm wrong

**Comment by coderman64** [comment]
Yeah. Looks interesting, but the process of integrating into the browser would be tough. 

Even so, any JS interpretation would be better than nothing. Netsurf has a built-in interpreter called "Duktape" that might be easier to implement. However, I haven't been able to get it to compile, so it isn't included in the current build. 

There are a few other things that I want to implement first, though. Particularly fonts.

**Comment by Moddimation** [comment]
I think the better idea would be to port duktape, who knows what limitations the existing one has? if it'll use the existing libraries it might end up hitting the same limitations as the official 3ds browser... not sure tho

**Comment by Anonymous941** [comment]
duktape is still not designed to interact with HTML in the way that websites require, but NetSurf does have [basic JavaScript support](https://www.netsurf-browser.org/documentation/info.html), so maybe that could be ported over:

> At present, NetSurf has only primitive and incomplete support for JavaScript, which we disable by default. Without JavaScript, NetSurf is able to provide access to most of sites on the World Wide Web. Some sites, however, will not display correctly or be unworkable due to heavy reliance on this standard.
> 
> Full JavaScript is support is planned but we are currently a long way off achieving it. Many things need to be done before complete JavaScript support can be considered, such as changing our web page layout engine to handle dynamic changes being made to the page. JavaScript support will require a huge amount of work to complete. If you are keen to help out please visit the [developer and contributor area](https://www.netsurf-browser.org/developers/).

**Comment by Trickiy** [comment]
If any Javascript is gonna be implamented [Wolfyxon's website](https://github.com/Wolfyxon/3ds-web-stuff) should be made compatible first.

**Comment by Anonymous941** [comment]
> If any Javascript is gonna be implamented [Wolfyxon's website](https://github.com/Wolfyxon/3ds-web-stuff) should be made compatible first.

The website you linked requires JavaScript, how is it possible to get support before it's added?

**Comment by Trickiy** [comment]
>The website you linked requires JavaScript, how is it possible to be supported before JavaScript is added?

I ment when Javascript is added.
